flake.nix

```diff
@@ -0,0 +1,187 @@
+{
+  description = "Build a cargo project";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    crane.url = "github:ipetkov/crane";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.rust-analyzer-src.follows = "";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    advisory-db = {
+      url = "github:rustsec/advisory-db";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    crane,
+    fenix,
+    flake-utils,
+    advisory-db,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+
+      inherit (pkgs) lib;
+      toolchain = fenix.packages.${system}.fromToolchainFile {
+        file = ./rust-toolchain.toml;
+        sha256 = "sha256-3jVIIf5XPnUU1CRaTyAiO0XHVbJl12MSx3eucTXCjtE=";
+      };
+
+      craneLib = (crane.mkLib pkgs).overrideToolchain toolchain;
+      src = craneLib.cleanCargoSource ./.;
+
+      # Common arguments can be set here to avoid repeating them later
+      commonArgs = {
+        inherit src;
+        strictDeps = true;
+
+        buildInputs = with pkgs;
+          [
+            libxkbcommon
+            xorg.libX11
+            vulkan-loader
+            vulkan-validation-layers
+            vulkan-headers
+          ]
```

Comment:
i ultimately couldn't get it to work on my machine because i kept needing gnome packages because i'm on ubuntu... but if it works for you, i think it's fine

Reply:
Can show me what it asked for? It shouldn't be too much of a hassle to add it in.
I might even be able to conditionally add the dependencies if the user has GNOME enabled.

Reply:
sure, the gnome-specific dependencies i needed were `gtk3 libsoup_3 webkitgtk_4_1`

Reply:
actually, i just tested and you only need `webkitgtk_4_1` since the other two are just dependencies